src/cloud-functions.ts

```diff
@@ -304,6 +312,41 @@ export interface MakeCloudFunctionArgs<EventData> {
   triggerResource: () => string;
 }
 
+/** @hidden */
+export function optionsToTrigger({
+  failurePolicy,
+  memory,
+  regions,
+  schedule,
+  timeoutSeconds,
+}: DeploymentOptions): TriggerAnnotated['__trigger'] {
+  const defaultFailurePolicy: FailurePolicy = {
+    retry: {},
+  };
+
+  const memoryLookup = {
+    '128MB': 128,
+    '256MB': 256,
+    '512MB': 512,
+    '1GB': 1024,
+    '2GB': 2048,
+  };
+
+  return {
```

Comment:
I think this one is more readable with multiple ifs as it was before, but @thechenky can weigh in on this one.

Reply:
I modified this code to make it easier to migrate from "everything optional" definition of `TriggerAnnotated['__trigger']` interface. I can change it back if needed, although I believe it would circle back with some refactor in the future, as TypeScript doesn't handle mutating objects in multiple stages very well.

Reply:
I agree with @kevinajian, the readability I feel was better before. I don't mind using ternary operator sparingly, where it's clear, like in `...(regions === undefined ? {} : { regions }),`, but I am having a really hard time understanding what is happening with failurePolicy here.

Reply:
I hope this helps with readability: 7b56f21194a9415b3158713e97c74335ecae3bf3

Reply:
I hope this helps with readability: 7b56f21194a9415b3158713e97c74335ecae3bf3.